### PR TITLE
resmgr: make sysfs/other hostfs pseudoroot directory configurable on the command line.

### PR DESCRIPTION
**Pull request by klihub**
Allow overriding the default `sysfs` mount point using the newly introduced `--sysfs-root $dir` command line option. When used, the real host sysfs should be available/mounted as `$dir/sys`.

Target usage for this is to allow an NRI-enabled `cri-resmgr` to run as  a `DaemonSet` inside a container, yet be able to do proper HW-discovery using the full `sysfs` of the host.

### CODE REVIEW

**Comment by askervin** [comment]
When making this configurable, wouldn't it be even more future-proof, if one could give sysfs mountpoint, defaulting to `/sys`, rather than giving a root directory and expecting the mountpoint to be called `sys` in it?

**Comment by klihub** [comment]
> When making this configurable, wouldn't it be even more future-proof, if one could give sysfs mountpoint, defaulting to `/sys`, rather than giving a root directory and expecting the mountpoint to be called `sys` in it?

When I looked into `pkg/topology` I realized that it has some correct but hard-coded assumptions about how an LFS-like sysfs hierarchy looks like, with support for an optional prefix (initially only for mocking sysfs for tests). Since it was the fastest/most trivial for me to modify configurability in `pkg/sysfs` to do the same, I opted to go that route. If you know of a setup where this would not work, please let me know and I can take another look.

**Comment by marquiz** [comment]
I think the current approach is actually better wrt. what @askervin suggested, as this is more general. It could be used for other system directories, too, if needed in the future. I'd just rename the option something else to reflect this. E.g. `--sys-root` or `--host-root` or smth. WDYT?

**Review by askervin — APPROVED** [review]
LGTM

File: pkg/sysfs/system.go [hunk 1]
```diff
@@ -30,9 +30,12 @@ import (
 	idset "github.com/intel/goresctrl/pkg/utils"
 )
 
+var (
+	// Sysfs root directory.
+	sysfsRoot = ""
```
Comment (marquiz) [inline pkg/sysfs/system.go:35]: nit: I think this is misleading here as it kinda represents the "host" root mount point. Rename to `sysRoot` or `hostRoot`, WDYT?

Comment (klihub) [reply pkg/sysfs/system.go:35]: Okay, true. I'll update.

Comment (klihub) [reply pkg/sysfs/system.go]: Updated taking that comment into account, and therefore also changing the function names to `SetSysRoot()` as opposed to the original `SetSysfsRoot()`.

**Review by marquiz — APPROVED** [review]
Thanks @klihub 
lgtm

**Pull request merged**